### PR TITLE
ORLP-195. Course rating unit test example.

### PR DESCRIPTION
**Pull request by antonbakalets**
This is an example of Controller unit test.

By the way, @vldmr1703, the unit test shows that your controller returns wrong links.

### CODE REVIEW

**Comment by vldmr1703** [comment]
@antonbakalets , problem is in getCourseRatingById()?

File: src/test/java/com/softserve/academy/spaced/repetition/controller/CourseRatingControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Test
+    public void getCourseRatingById() throws Exception {
+        when(courseRatingService.getCourseRatingById(eq(5L))).thenReturn(createCourseRating());
+        mockMvc.perform(get("/api/rate/course/{id}", 5L)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("{\"rating\":3,\"accountEmail\":\"email@email\",\"courseId\":5,\"links\":[{\"rel\":\"self\",\"href\":\"http://localhost/api/rate/course/77\"}]}"));
```
Comment (vldmr1703) [inline src/test/java/com/softserve/academy/spaced/repetition/controller/CourseRatingControllerTest.java]: Instead of "links" should be "_links", isn't it?

Comment (b0hdan) [reply src/test/java/com/softserve/academy/spaced/repetition/controller/CourseRatingControllerTest.java]: Yes, I agree.

Comment (b0hdan) [reply src/test/java/com/softserve/academy/spaced/repetition/controller/CourseRatingControllerTest.java]: Oh, and, besides, you have forgotten to add the port here.

Comment (antonbakalets) [reply src/test/java/com/softserve/academy/spaced/repetition/controller/CourseRatingControllerTest.java]: Doesn't matter the port - for the test it is sufficient.

The problem is that course has id=5, but link is "api/rate/course/77", I suspect this is wrong.

Comment (vldmr1703) [reply src/test/java/com/softserve/academy/spaced/repetition/controller/CourseRatingControllerTest.java]: @antonbakalets, but this method returns the rating by its id, not by id of course.

**Pull request merged**